### PR TITLE
Change notification mails

### PR DESCRIPTION
**Pull request by jongilbraith**
* Can yank the letter opener commit if you're not keen on it.
* I didn't make the constant in `Appointment` an array of symbols as mentioned as it's being used to compare with `appointment.previous_changes.keys` which expects strings.
* The new email behaviour is covered by additional scenarios.
* The move of the cancelled/confirmed/missed mails into the job is covered by existing features.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -69,6 +69,7 @@ source 'https://rubygems.org' do
   end
 
   group :development do
+    gem 'letter_opener'
```
Comment (benlovell) [inline Gemfile]: Can we remove this in favour of a mailer preview instead? This follows with the rest of our emails.

File: app/jobs/appointment_cancelled_notification_job.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+class AppointmentCancelledNotificationJob < ActiveJob::Base
+  queue_as :default
+
+  MESSAGE = 'cancelled'.freeze
```
Comment (benlovell) [inline app/jobs/appointment_cancelled_notification_job.rb]: I think these message constants should be defined in the `CustomerUpdateActivity`. Any overriding reason they're defined here?

File: app/jobs/appointment_confirmed_notification_job.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+class AppointmentConfirmedNotificationJob < ActiveJob::Base
```
Comment (benlovell) [inline app/jobs/appointment_confirmed_notification_job.rb]: Thinking about these job classes, maybe we should just collapse them and pass the `message` as an argument. We could then just `case` over the `message` and invoke the right there. The activity creation and push notification stuff would just work as they're common across all three. WDYT?

File: app/lib/notifier.rb [hunk 4]
```diff
@@ -23,17 +23,18 @@ def guiders_to_notify(appointment)
...
     elsif appointment_missed?(appointment)
-      AppointmentMailer.missed(appointment).deliver_later
+      AppointmentMissedNotificationJob.perform_later(appointment)
     end
   end
 
-  def appointment_time_changed?(appointment)
-    appointment.previous_changes.slice('start_at', 'end_at').present?
+  def appointment_details_changed?(appointment)
+    appointment.previous_changes.any? &&
```
Comment (benlovell) [inline app/lib/notifier.rb]: WDYT about moving this to a predicate such as `Appointment#notify?` Then we could just call that in the `if` statement in `notify_customer`.

File: Gemfile (hunk 1 above)
Comment (jongilbraith) [inline Gemfile]: 👍

File: app/jobs/appointment_cancelled_notification_job.rb (hunk 2 above)
Comment (jongilbraith) [inline app/jobs/appointment_cancelled_notification_job.rb]: Yep, makes sense.

File: app/lib/notifier.rb (hunk 4 above)
Comment (jongilbraith) [inline app/lib/notifier.rb]: And do the same thing with `appointment_cancelled?` and `appointment_missed?` for consistency?

Comment (benlovell) [inline app/lib/notifier.rb]: Yeah good shout. Perhaps `notify_missed?` and `notify_cancelled?`?

File: app/jobs/appointment_confirmed_notification_job.rb (hunk 3 above)
Comment (jongilbraith) [inline app/jobs/appointment_confirmed_notification_job.rb]: Yeah, certainly would be nice to cut down on the boilerplate. I'll do a quick rejig and see how it looks.

File: app/lib/notifier.rb (hunk 4 above)
Comment (jongilbraith) [inline app/lib/notifier.rb]: 👍

**Comment by benlovell** [comment]
[As a user when I update certain customer details an updated confirmation email is sent to the customer](https://trello.com/c/JNDAAOYR/30-as-a-user-when-i-update-certain-customer-details-an-updated-confirmation-email-is-sent-to-the-customer)

**Pull request merged**